services/implementations/DaoStudentEntryService.java

```diff
@@ -0,0 +1,34 @@
+package services.implementations;
+
+import data.daos.StudentDao;
+import data.daos.implementations.ArrayListStudentDao;
+import logic.models.Student;
+import services.StudentEntryService;
+
+public class DaoStudentEntryService implements StudentEntryService {
```

Comment:
Yes, it is hard to find a meaningful name that sticks to a consistency, or patterns and also reveal intention of what the class, method, property, argument, or variable does. In this case a common convention in the Java world for naming would be `StudentServiceImpl` since it is the implementation of the `StudentService` interface. It does reveal that it is the implementation and not the interface, but I also agree that this is [not a very good name](https://octoperf.com/blog/2016/10/27/impl-classes-are-evil/). It is also good to stick to conventions if there are any, and each decision has a trade off. We still use an interface to do for example dependency injection but most important to pass [mock objects](https://www.vogella.com/tutorials/Mockito/article.html) (dummy implementation of interfaces) for testing (e.g. unit tests).

Reply:
Updated.